src/engraving/layout/v0/tlayout.cpp

```diff
@@ -3634,9 +3634,7 @@ void TLayout::layout(Ornament* item, LayoutContext& ctx)
         return;
     }
 
-    Note* cueNote = cueNoteChord->notes().front();
-    ChordLayout::layoutChords3(ctx.conf().style(), { cueNoteChord }, { cueNote }, item->staff(), ctx);
-    layout(cueNoteChord, ctx);
+    // CAUTION: assumes that cueNoteChord has been already laid out
```

Comment:
Can you call TLayout::layoutTrillCueNote here? This way we will have clearer dependencies in the code and won't need this comment

Reply:
Unfortunately no, because it shouldn't _always_ be called (that's what was causing the crash). Basically, the Ornament layout gets called from two different places, one [here](https://github.com/mike-spa/MuseScore/blob/master/src/engraving/layout/v0/chordlayout.cpp#L1861C13-L1861C44) and the other [here](https://github.com/mike-spa/MuseScore/blob/master/src/engraving/layout/v0/chordlayout.cpp#L1003C13-L1003C37). In the first case, we should also layout the trill cue note, but in the second case we shouldn't (this was the cause of the crash). So I decided to split that part into its own separate function, and call it separately. The only other option would be to pass some bool flag here, which decides whether to call it or not, but that means I have to change the function signature of two layout functions (Articulation and Ornament), so I thought this way worked better.

Reply:
We should not call layoutTrillCueNote here, but have to perform other manipulations on the cueNoteChord geometry (like setPosX)? This is very confusing. Also, this comment is even more misleading because it assumes that we always call layoutTrillCueNote

1. all layout-related code for the cue note should be in one place (layoutTrillCueNote)
2. it should be explicitly called where it's needed and you should avoid such comments assuming that it has already been called somewhere

Reply:
ok, should be better now 👍